crates/symbolicator/src/services/cacher.rs

```diff
@@ -190,74 +215,67 @@ pub trait CacheItemRequest: 'static + Send + Sync + Clone {
 }
 
 impl<T: CacheItemRequest> Cacher<T> {
-    /// Constructs the path corresponding to the given [`CacheKey`].
-    ///
-    /// Returns [`None`] if caching is disabled.
-    fn get_cache_path(&self, key: &CacheKey, version: u32) -> Option<PathBuf> {
-        let mut cache_dir = self.config.cache_dir()?.to_path_buf();
-
-        if version != 0 {
-            cache_dir.push(version.to_string());
-        }
-
-        cache_dir.push(safe_path_segment(key.scope.as_ref()));
-        cache_dir.push(safe_path_segment(&key.cache_key));
-
-        Some(cache_dir)
-    }
-
     /// Look up an item in the file system cache and load it if available.
     ///
     /// Returns `Ok(None)` if the cache item needs to be re-computed, otherwise reads the
     /// cached data from disk and returns the cached item as returned by
     /// [`CacheItemRequest::load`].
     ///
+    /// Calling this function when the cache is not enabled will simply return `Ok(None)`.
+    ///
     /// # Errors
     ///
     /// If there is an I/O error reading the cache [`CacheItemRequest::Error`] is returned.
-    fn lookup_cache(
+    fn lookup_local_cache(
         &self,
         request: &T,
         key: &CacheKey,
-        path: &Path,
+        version: u32,
     ) -> Result<Option<T::Item>, T::Error> {
-        let name = self.config.name();
-        log::trace!("Trying {} cache at path {:?}", name, path);
-        sentry::with_scope(
-            |scope| {
-                scope.set_extra(
-                    &format!("cache.{}.cache_path", name),
-                    format!("{:?}", path).into(),
-                );
-            },
-            || {
-                let byteview = match self.config.open_cachefile(path)? {
-                    Some(x) => x,
-                    None => return Ok(None),
-                };
-
-                let status = CacheStatus::from_content(&byteview);
-                if status == CacheStatus::Positive && !request.should_load(&byteview) {
-                    log::trace!("Discarding {} at path {:?}", name, path);
-                    metric!(counter(&format!("caches.{}.file.discarded", name)) += 1);
-                    return Ok(None);
-                }
-
-                // This is also reported for "negative cache hits": When we cached the 404 response from a
-                // server as empty file.
-                metric!(counter(&format!("caches.{}.file.hit", name)) += 1);
-                metric!(
-                    time_raw(&format!("caches.{}.file.size", name)) = byteview.len() as u64,
-                    "hit" => "true"
-                );
-
-                let path = path.to_path_buf();
-                log::trace!("Loading {} at path {:?}", name, path);
-                let item =
-                    request.load(key.scope.clone(), status, byteview, CachePath::Cached(path));
-                Ok(Some(item))
-            },
-        )
+        match self.config.cache_dir() {
+            Some(ref cache_dir) => {
```

Comment:
Lets early return on `None` here, to avoid excessive rightward drift.

Reply:
meh, i'm not a huge fan of early returns...